fzf-git.sh

```diff
@@ -140,8 +140,16 @@ if [[ $1 = --list ]]; then
     fi
 
     case "$(uname -s)" in
-      Darwin) open "$url$path"     ;;
-      *)      xdg-open "$url$path" ;;
+      Darwin)
+        open "$url$path"
+        ;;
+      *)
```

Comment:
Would it be possible to reliably detect WSL using `uname`? Do you think checking `$WSL_DISTRO_NAME` is better?

Reply:
Yeah, I agree - how much condition checking is appropriate for WSL detection is definitely the most controversial part of this issue.

Initially, I tried checking the existence of `/proc/sys/fs/binfmt_misc/WSLInterop`, but found it unclear and not entirely reliable. For example, I noticed that in environments like a `podman-machine` created by `podman-desktop`, this file wasn't even present.

If I were to add more checks for this feature, I'd personally consider:

- Checking the suffix of `uname -a`, like `5.15.167.4-microsoft-standard-WSL2`
- Ensuring `explorer.exe` exists using `command -v explorer.exe`

To me, combining these three checks would still be acceptable. What do you think sounds better?

Reply:
I was thinking of simplifying the conditional branches only using the output of `uname` like so:

```sh
case "$(uname -sr)" in
  *Darwin*)
    open
    ;;
  *microsoft* | *Microsoft*)
    explorer
    ;;
  *)
    xdg-open
    ;;
esac
```

Reply:
I was initially concerned that extracting `microsoft` might result in a false positive on Azure Linux. However, after testing with the [Azure Linux 3.0 image](https://github.com/microsoft/azurelinux/blob/7e483a893e4413d0002bfc3080f7175e4b371289/README.md?plain=1#L24), it seems to be safe at least in that case.

![441037691-117ace8c-04af-42a2-9f96-588a905f91f9](https://github.com/user-attachments/assets/728594cb-bf1f-491f-83b3-9500fb9cf4cc)

On WSL2
```console
> uname -sr
Linux 5.15.167.4-microsoft-standard-WSL2
```